packages/propeller/src/Array.ts

```diff
@@ -0,0 +1,10 @@
+export function keepMap<T, V>(arr: T[], fn: (t: T) => V | undefined): V[] {
+  const result = new Array<V>(arr.length)
+  let idx = 0
+  for (let i = 0; i < arr.length; i++) {
+    const v = fn(arr[i])
+    if (v) result[idx++] = v
```

Comment:
could delay copying until first item is being dropped, or would you normally expect to be dropping something?

Reply:
the most common case seems to be that it's called with one empty array and one array with some items. Mostly because overriding the write position is a `merge` with an empty queue